surround/runners.py

```diff
@@ -42,10 +41,29 @@ def __init__(self, assembler):
         self.assembler = assembler
 
     @abstractmethod
-    def run(self, is_training=False):
+    def load_data(self, mode, config):
+        """
+        Load the data and prepare it to be fed into the :class:`surround.assembler.Assembler`.
+
+        :param mode: the mode the assembly was run in (batch, train, predict, web)
+        :type mode: :class:`surround.runners.RunMode`
+        :param config: the configuration of the assembly
+        :type config: :class:`surround.config.Config`
+        """
+
+    def run(self, mode=RunMode.BATCH_PREDICT):
         """
         Prepare data and execute the :class:`surround.assembler.Assembler`.
 
         :param is_training: Run the pipeline in training mode or not
         :type is_training: bool
         """
+
+        # Initialise the assembler
+        self.assembler.init_assembler(mode == RunMode.BATCH_PREDICT)
+
+        # Load the data to be fed into the assembler
+        data = self.load_data(mode, self.assembler.config)
```

Comment:
Check that the data object here is an instance of `State` here so that we minimise developer errors. This may be something you can set up once we have the type hints PR merged...

Reply:
Just added a check for now, done :)